roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -94,7 +120,30 @@ def subclasses(cls):
             for subclass in subclass.subclasses():
                 yield subclass
 
-    def execute_module(self, module_name=None, module_args=None):
+    def register_failure(self, error):
+        """Record failure in the check, to be merged into result hash and saved to output directory."""
+        # It should be an exception; make it one if not
+        if not isinstance(error, OpenShiftCheckException):
+            error = OpenShiftCheckException(str(error))
```

Comment:
Hmm, why is this done?

Reply:
There was one place in the action plugin where it takes the hash result and turns `failed` + `msg` into a registered failure, and I thought it would be simpler to make that generic, to ensure that what's in `failures` is always an `OpenShiftCheckException`.